x-pack/plugins/entity_manager/server/routes/entities/search.ts

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import moment from 'moment';
+import { z } from '@kbn/zod';
+import { createEntityManagerServerRoute } from '../create_entity_manager_server_route';
+import { entitySourceSchema } from '../../lib/queries';
+
+export const searchEntitiesRoute = createEntityManagerServerRoute({
+  endpoint: 'POST /internal/entities/_search',
+  params: z.object({
```

Comment:
So I cannot (yet) pass in any filters or metadata I want?

Reply:
right you can at the source level but not type level, I can add that

Reply:
I mean at the request level, if for a certain request I only want entities that match a certain filter or I want them to also include some other metadata field (not listed in the source), can I do that?